### PR TITLE
Wallet transactions export

### PR DESCRIPTION
**Pull request by dennisreimann**
Adds an export (CSV and JSON) to the wallet transactions, similar to the invoices export.

The exported data needs some more work, but in general this is mostly done. Feedback welcome!

Wallet transactions export

### CODE REVIEW

**Comment by dstrukt** [comment]
Nice, great first steps toward feature and functionality pairing!

Enables us to address the next set of updates across all views equally.

File: BTCPayServer/Controllers/UIWalletsController.cs [hunk 1]
```diff
@@ -1278,6 +1279,39 @@ await ExplorerClientProvider.GetExplorerClient(walletId.CryptoCode)
             }
         }
 
+        [HttpGet("{walletId}/export")]
+        [Authorize(AuthenticationSchemes = AuthenticationSchemes.Cookie, Policy = Policies.CanViewInvoices)]
```
Comment (NicolasDorier) [inline BTCPayServer/Controllers/UIWalletsController.cs:1283]: I don't think CanViewInvoices should have this right, we should keep it to `Policies.CanModifyStoreSettings` (already required at controller level)

Comment (NicolasDorier) [inline BTCPayServer/Controllers/UIWalletsController.cs:1283]: I don't think CanViewInvoices should have this right, we should keep it to `Policies.CanModifyStoreSettings` (already required at controller level)

**Comment by NicolasDorier** [comment]
This need a small selenium test that click on json/csv. Can modify an existing test.

**Comment by NicolasDorier** [comment]
This need a small selenium test that click on json/csv. Can modify an existing test.

File: BTCPayServer/Controllers/UIWalletsController.cs (hunk 1 above)
Comment (dennisreimann) [reply BTCPayServer/Controllers/UIWalletsController.cs]: Fixed in [32d6677](https://github.com/btcpayserver/btcpayserver/pull/3744/commits/32d66771c09cd219309654cd2965911610d9e3b2).

**Comment by dennisreimann** [comment]
Test cases added and the export format is finalized now. Ready for review :)

**Comment by NicolasDorier** [comment]
@dennisreimann I understand selenium discourage download. However, you still need to exercise the code paths.
You can user the tester and instantiate the controller directly and call it if your prefer.

**Comment by dennisreimann** [comment]
@NicolasDorier Updated it, not sure if this is what you wanted though. Feel free to change it further :)

**Comment by dennisreimann** [comment]
Ok, it looks like the CSV download check doesn't work in CI anyways, removed it.

**Pull request merged**